### PR TITLE
#8158: Update heaviside sweep config and doc

### PR DESCRIPTION
**Pull request by mcw-anasuya**
### Ticket
#8158 

### Problem description
Documentation missing for Heaviside op with supported params

### What's changed
Updated doc and corresponding sweep test files

### Checklist
- [x] [Post commit CI passes](https://github.com/tenstorrent/tt-metal/actions/runs/11362041107) - PASSED
- [x] [Run Sweeps](https://github.com/tenstorrent/tt-metal/actions/runs/11347817020) - PASSED

### CODE REVIEW

File: tests/ttnn/sweep_tests/sweeps/sweeps/heaviside.py [hunk 1]
```diff
@@ -16,14 +16,29 @@
...
+    height,
+    width,
+    input_dtype,
+    input_memory_config,
+    output_memory_config,
+    layout,
+    value,
+) -> Tuple[bool, Optional[str]]:
+    if layout == ttnn.ROW_MAJOR_LAYOUT or input_dtype == ttnn.bfloat8_b:
+        return True, "Skipped since ROW_MAJOR_LAYOUT is not supported and test fails for bfloat8_b"
```
Comment (Aswinmcw) [inline tests/ttnn/sweep_tests/sweeps/sweeps/heaviside.py:38]: Test fail means pcc drop or something else? Juse not supported is not enough?

**Review by Aswinmcw — APPROVED** [review]
Minor comment, LGTM

**Review by VirdhatchaniKN — APPROVED** [review]
LGTM

**Review by KalaivaniMCW — APPROVED** [review]

File: tests/ttnn/sweep_tests/sweeps/sweeps/heaviside.py (hunk 1 above)
Comment (umadevimcw) [reply tests/ttnn/sweep_tests/sweeps/sweeps/heaviside.py:38]: Is it bf8 supported?

File: ttnn/cpp/ttnn/operations/eltwise/unary/unary_pybind.hpp [hunk 2]
```diff
@@ -1390,7 +1391,13 @@ void py_module(py::module& module) {
     // Unaries with float parameter
     detail::bind_unary_operation_with_float_parameter(module, ttnn::elu, "alpha", "The alpha parameter for the ELU function", "");
     detail::bind_unary_operation_with_float_parameter(module, ttnn::rsub, "value", "subtrahent value which is actually calculated as minuend", "Returns tensor with respective elements of the input tensor subtracted from the value.");
-    detail::bind_unary_operation_with_float_parameter(module, ttnn::heaviside, "value", "The value parameter for the Heaviside function", "");
+    detail::bind_unary_operation_with_float_parameter(module, ttnn::heaviside, "value", "The value parameter for the Heaviside function",
+    R"doc(Supported dtypes, layouts, and ranks:
+        +----------------------------+---------------------------------+-------------------+
+        |     Dtypes                 |         Layouts                 |     Ranks         |
+        +----------------------------+---------------------------------+-------------------+
```
Comment (umadevimcw) [inline ttnn/cpp/ttnn/operations/eltwise/unary/unary_pybind.hpp:1398]: Update it if Bf8 is not supported

**Review by umadevimcw — CHANGES_REQUESTED** [review]

File: tests/ttnn/sweep_tests/sweeps/sweeps/heaviside.py (hunk 1 above)
Comment (mcw-anasuya) [reply tests/ttnn/sweep_tests/sweeps/sweeps/heaviside.py:38]: It is failing (not crashing) for bf8_b. So I have updated the pybind to support only bf16.

**Review by umadevimcw — APPROVED** [review]

**Review by eyonland — APPROVED** [review]

**Review by jdesousa-TT — APPROVED** [review]

File: tests/ttnn/sweep_tests/sweeps/sweeps/heaviside.py (hunk 1 above)
Comment (mcw-anasuya) [reply tests/ttnn/sweep_tests/sweeps/sweeps/heaviside.py]: @umadevimcw @Aswinmcw 
bf8_b is now passing with high pic. I have updated the document accordingly.

**Review by umadevimcw — APPROVED** [review]

**Pull request merged**